upload_log_files.py

```diff
@@ -0,0 +1,157 @@
+    files_list = [file for file in os.listdir(dir_path) if file.endswith((".gzip", ".profile"))]
+    file_paths = [os.path.join(dir_path, basename) for basename in files_list]
+
+    return file_paths
+
+
+def get_uploaded_file_dates(uploaded_files_list, date_pattern):
+    dates_match = [re.search(date_pattern, file) for file in uploaded_files_list]
+    uploaded_file_dates = []
+    for date_match in dates_match:
+        if date_match == None:
+            continue
+        uploaded_file_dates.append(date_match.group())
+
+    return uploaded_file_dates
+
+
+def get_files_by_date(dir_path, uploaded_file_dates):
+    file_paths = get_file_paths(dir_path)
+    files_by_date = {}
+    if len(file_paths) > 0:
+        for file in file_paths:
+            file_date_match = re.search(date_pattern, file)
+            file_date = file_date_match.group()
+            if file_date in uploaded_file_dates:
+                log.info(f" file already uploaded for {file_date}, skipping...")
+            else:
+                if file_date not in files_by_date:
+                    files_by_date[file_date] = []
+                files_by_date[file_date].append(file)
+    else:
+        log.info(f" No file found in {dir_path}!, skipping...")
+
+    return files_by_date
+
+
+def delete_old_log_files(dir_path, uploaded_file_dates):
+    log_file_paths = get_file_paths(dir_path)
+    files_for_days_that_upload_failed = {}
+
+    most_recent_file_path = None
+    if len(log_file_paths) > 0:
+        most_recent_file_path = max(log_file_paths, key=os.path.getmtime)
+
+    for file_path in log_file_paths:
+        file_date_match = re.search(date_pattern, file_path)
+        file_date = file_date_match.group()
+
+        # Create a list of files for days that failed to upload
+        if file_date in uploaded_file_dates:
+            if file_path == most_recent_file_path:
+                log.info(f"Retaining latest modified file {file_path} for quick retrieval")
+                continue
+
+            log.warning(f"Deleting {file_path} because files for {file_date} already uploaded to cloud")
+            os.remove(os.path.join(dir_path, file_path))
+
+        # Delete files for days that have a file uploaded in g-cloud
+        else:
+            log.debug(f'Files for {file_date} not yet uploaded to cloud, '
+                      f'will delete other files and retain the latest modified file for upload')
+            if file_date not in files_for_days_that_upload_failed:
+                files_for_days_that_upload_failed[file_date] = []
+
+            files_for_days_that_upload_failed[file_date].append(file_path)
+
+    # Check for latest modified file path for each day that failed to upload
+    # Delete other files for that date
+    for file_date in files_for_days_that_upload_failed:
+        for file_path in files_for_days_that_upload_failed[file_date]:
+            if file_path == max(files_for_days_that_upload_failed[file_date], key=os.path.getmtime):
```

Comment:
```suggestion
            if file_path == most_recent_file_path:
```

Reply:
to fix the issue identified under Imaqal covid19 https://github.com/AfricasVoices/Project-IMAQAL-COVID19/pull/25/files